In [0]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt


In [0]:
sale_data={'SalesPerson':[1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30],
            'OldScheme':[57,103,59,75,84,73,35,110,44,82,67,64,78,53,41,39,80,87,73,65,28,62,49,84,63,77,67,101,91,50],
            'NewScheme':[62,122,54,82,84,86,32,104,38,107,84,85,99,39,34,58,73,53,66,78,41,71,38,95,81,58,75,94,100,68]
          }
sale_pd = pd.DataFrame(sale_data)
sale_pd

SalesPerson  OldScheme  NewScheme
0             1         57         62
1             2        103        122
2             3         59         54
3             4         75         82
4             5         84         84
5             6         73         86
6             7         35         32
7             8        110        104
8             9         44         38
9            10         82        107
10           11         67         84
11           12         64         85
12           13         78         99
13           14         53         39
14           15         41         34
15           16         39         58
16           17         80         73
17           18         87         53
18           19         73         66
19           20         65         78
20           21         28         41
21           22         62         71
22           23         49         38
23           24         84         95
24           25         63         81
25           26         77         58
26           27         67         75
27           28        101         94
28           29         91        100
29           30         50         68

In [0]:
#set index with person id and drop column
sale_pd.set_index(sale_pd.SalesPerson)
sale_pd=sale_pd.drop(columns='SalesPerson')
sale_pd

OldScheme  NewScheme
0          57         62
1         103        122
2          59         54
3          75         82
4          84         84
5          73         86
6          35         32
7         110        104
8          44         38
9          82        107
10         67         84
11         64         85
12         78         99
13         53         39
14         41         34
15         39         58
16         80         73
17         87         53
18         73         66
19         65         78
20         28         41
21         62         71
22         49         38
23         84         95
24         63         81
25         77         58
26         67         75
27        101         94
28         91        100
29         50         68

In [0]:
#using describe function to see spreading data,mean max,range ,Quartile
sale_pd.describe()

OldScheme   NewScheme
count   30.000000   30.000000
mean    68.033333   72.033333
std     20.455980   24.062395
min     28.000000   32.000000
25%     54.000000   55.000000
50%     67.000000   74.000000
75%     81.500000   85.750000
max    110.000000  122.000000

initial analysis : Mean of  OldScheme and NewScheme is 68.03 and 72.03 respectively and median is 67 and 74 meaning mean of New Scheme has move to slightly lower side from median implies that have some Outlier at lower side and also min and max get increased at lower 

Gap-margin between Q2 and Q3 become lower than gap between Q1 and Q2 it mean some sellers does increase selling but not seeing higher marginal growth in sellings.


In [0]:
#analysis five percent significance test
new_mean = sale_pd.NewScheme.mean()
new_mean

72.03333333333333

In [0]:
old_mean = sale_pd.OldScheme.mean()
old_mean

68.03333333333333

In [0]:
#draw normal distribution curve 


Problem : 2 
Consinder there is no difference between sample mean before scheme and after scheme sale \
H0 : mean_newScheme - mean_oldScheme =0 \
Ha : mean_newScheme > mean_oldScheme (upper tail (one tail t test))

In [0]:
# although it is dependent two sample test
# we need to comapare mean before event and after event
from scipy import stats
#t_stats,p_value=stats.ttest_rel(sale_pd.OldScheme,sale_pd.NewScheme)
t_stats,p_value=stats.ttest_rel(sale_pd.NewScheme,sale_pd.OldScheme)

In [0]:
# as per reading API document ttest_rel returns two sided tail p_value
# we need to devide by 2 
p_value=p_value/2
print("P-value : ",p_value)

P-value :  0.06528776980668831


In [0]:
print("T-stat : ",t_stats)

T-stat :  1.5559143823544377


Problem : 3  Since P value is 0.065 which is greater than alpha significant value 0.05 thats mean accepting null hypothesis or rejects alternate hypothesis which results new scheme does not raised output significantly.

Problem : 4a
a ) Type I error occurs when a significance test results in the rejection of a true null hypothesis so rejection falls in 0.05 region in null hypothesis.Probability of type 1 error is 0.05 \

Problem : 4b  

      value of difference of mean at significant level which is critical value at which null hypothesis is going to accepted or rejected.
    t value at 0.05 significant level of alpha is 1.699 for degree of freedom 29 (here sample size is 30 then DF = 30-1=29)
    though we know t_statics = (X-mean)/standard_Error 
    here difference at center is zero of two group then equation become X = t_statics*standard_error
    SE = meanDifference/t_statics ==> here mean_difference of two scheme is 4000 and t_statics value is 1.5559143823544377
    then critical value = (t value at 0.05) 

In [0]:
SE = 4000/t_stats
Difference_of_mean_at_CV = (stats.t.ppf(1-0.05,29))*SE # (1-0.05) for upper tail
Difference_of_mean_at_CV

4368.176156228718

In [0]:
# now calculation of t_statics value at point where difference is 5000
t_stat_at_5000 = 5000/SE

In [0]:
# probability to difference is going to greater than 5000 is 
p_value_at_greaterfrom5000 =(1-stats.t.cdf(t_stat_at_5000,29))
p_value_at_greaterfrom5000

0.030771084315083463

Problem 4c : Power of test

In [0]:
# power of test is not occuring Type-2 error if B is type-2 error then power of test is (1-B)
# when alternate hyposthesis is true and mean difference is 5000
ud=5000
x=Difference_of_mean_at_CV #(alpha critical value)
t = (4368.176-5000)/SE
t

-0.24576601217917743

In [0]:
# Beta calculation
beta = stats.t.cdf(t,29)
beta

0.4037972291824633

In [0]:
#power of test
power_test = 1-beta
power_test

0.5962027708175367